### PR TITLE
mynewt-newt: 1.3.0 -> 1.7.0, use buildGoModule

### PR DESCRIPTION
**Pull request by Thra11**
###### Motivation for this change
Update mynewt-newt to 1.7.0. Switch from legacy buildGoPackage to buildGoModule

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Thra11** [comment]
Looks like 1.7.0 has a bug which prevents `newt upgrade` working with mcuboot (Unreleased workaround/fix is here https://github.com/apache/mynewt-newt/commit/7d4ef3fe65a9a83cc58e7bd973654ad235cc68bc). Not sure if we should patch it or just wait for a release...

**Comment by kalbasit** [comment]
@Thra11 Does 1.7.0 work well with the patch applied? It's probably ok to update it with the patch if it works well.

**Comment by Thra11** [comment]
I had to cherry pick an additional commit for the fix to apply cleanly. Having done that, it builds and seems to work. I tried
1. Removing my existing mynewt installation
2. Switching to the 0.3.0 version of the package and running `newt install`
3. Switching to the patched 0.7.0 version of the package and running `newt upgrade`
It seems to have worked ok, where the unpatched version failed.

**Review by kalbasit — APPROVED** [review]

**Review by marsam — CHANGES_REQUESTED** [review]

File: pkgs/tools/package-management/mynewt-newt/default.nix [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
-    repo = "incubator-mynewt-newt";
+    repo = "mynewt-newt";
     rev = "mynewt_${builtins.replaceStrings ["."] ["_"] version}_tag";
-    sha256 = "0ia6q1wf3ki2yw8ngw5gnbdrb7268qwi078j05f8gs1sppb3g563";
+    sha256 = "0rwn4ghh7kal8csxlh0w1p29b5m1nam9lkrxla5wdfhnzbsg8hfa";
   };
 
+  patches = [
+    ./0001-Use-detached-head-instead-of-local-branches.patch
+    ./0002-Mcuboot-hack-to-prevent-newt-upgrade-failure.patch
```
[the remark this answers is not in the document]
Comment (Thra11) [reply pkgs/tools/package-management/mynewt-newt/default.nix]: Done. Thanks for the review!

**Comment by kalbasit** [comment]
@GrahamcOfBorg build mynewt-newt

**Review by kalbasit — APPROVED** [review]

**Pull request merged**